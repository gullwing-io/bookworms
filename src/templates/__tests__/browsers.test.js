import { jest } from "@jest/globals";
import {
  bookmark,
  bookmarkFolder,
  bookmarkLink,
  addDescription,
  addBookwormsDescription,
} from "../browsers";

// I don't like using snapshots for these tests but because of the HTML formatting its easier
describe("Templates - browsers", () => {
  describe("bookmark", () => {
    test("should return structure for browsers bookmarks", () => {
      expect(
        bookmark(
          "26/09/2021 - 6pm",
          "Bookmarks label",
          "Bookmarks description",
          "<div>Children HTML</div>"
        )
      ).toMatchSnapshot();
    });
  });

  describe("bookmarkFolder", () => {
    test("should return description, header and children", () => {
      expect(
        bookmarkFolder(
          "26/09/2021 - 6pm",
          "Section label",
          "Section description",
          "<div>Children HTML</div>"
        )
      ).toMatchSnapshot();
    });
  });

  describe("bookmarkFolder", () => {
    test("should return description, header and children", () => {
      expect(
        bookmarkFolder(
          "26/09/2021 - 6pm",
          "Link label",
          "Link description",
          "<div>Children HTML</div>"
        )
      ).toMatchSnapshot();
    });
  });

  describe("bookmarkLink", () => {
    test("should return description and link", () => {
      expect(
        bookmarkLink(
          "26/09/2021 - 6pm",
          "Link label",
          "Link description",
          "https://www.testlink.com"
        )
      ).toMatchSnapshot();
    });
  });

  describe("addDescription", () => {
    test("should return description comment", () => {
      expect(addDescription("hello world")).toEqual("<!-- hello world -->");
    });
    test("should return empty string", () => {
      expect(addDescription("")).toEqual("");
    });
  });

  describe("addBookwormsDescription", () => {
    test("should return description for file comment", () => {
      jest
        .spyOn(Date.prototype, "toLocaleString")
        .mockReturnValue("30/09/2021, 19:43:14");
      expect(addBookwormsDescription(1633027394454)).toEqual(
        "<!-- These bookmarks were last updated on 30/09/2021, 19:43:14 using bookworms (https://github.com/thearegee/bookworms) -->"
      );
    });
  });
});
