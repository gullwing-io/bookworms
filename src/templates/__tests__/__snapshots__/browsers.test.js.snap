// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Templates - browsers bookmark should return structure for browsers bookmarks 1`] = `
"
    <!-- These bookmarks were last updated on Invalid Date using bookworms (https://github.com/thearegee/bookworms) -->
    <!DOCTYPE NETSCAPE-Bookmark-file-1>
<!-- This is an automatically generated file.
        It will be read and overwritten.
        DO NOT EDIT! -->
<META HTTP-EQUIV="Content-Type" CONTENT="text/html; charset=UTF-8">
<TITLE>Bookmarks</TITLE>
<H1>Bookmarks</H1>
<!-- Bookmarks description -->
<DL><p>
<DT><H3 ADD_DATE="26/09/2021 - 6pm" LAST_MODIFIED="26/09/2021 - 6pm">Bookmarks label</H3>
<DL><p>
    <div>Children HTML</div>
</DL><p>
</DL><p>
    "
`;

exports[`Templates - browsers bookmarkFolder should return description, header and children 1`] = `
"
            <!-- Section description -->
            <DT><H3 ADD_DATE="26/09/2021 - 6pm" LAST_MODIFIED="26/09/2021 - 6pm">Section label</H3>
<DL><p>
<div>Children HTML</div>
</DL><p>
    "
`;

exports[`Templates - browsers bookmarkFolder should return description, header and children 2`] = `
"
            <!-- Link description -->
            <DT><H3 ADD_DATE="26/09/2021 - 6pm" LAST_MODIFIED="26/09/2021 - 6pm">Link label</H3>
<DL><p>
<div>Children HTML</div>
</DL><p>
    "
`;

exports[`Templates - browsers bookmarkLink should return description and link 1`] = `
"
    <!-- Link description -->
    <DT><A HREF="https://www.testlink.com" ADD_DATE="26/09/2021 - 6pm">Link label</A>
    "
`;
